public/app/features/dashboard/components/DashNav/DashNav.tsx

```diff
@@ -252,42 +251,41 @@ class DashNav extends PureComponent<Props> {
 
     if (showSettings) {
       buttons.push(
-        <ToolbarButton tooltip="Dashboard settings" icon="cog" onClick={this.onOpenSettings} key="button-settings" />
+        <ToolbarButton tooltip="Dashboard settings" icon="cog" onClick={onOpenSettings} key="button-settings" />
       );
     }
 
-    this.addCustomContent(customRightActions, buttons);
+    addCustomContent(customRightActions, buttons);
 
-    buttons.push(this.renderTimeControls());
+    buttons.push(renderTimeControls());
     buttons.push(tvButton);
     return buttons;
-  }
+  };
 
-  gotoSnapshotOrigin(snapshotUrl: string) {
+  const gotoSnapshotOrigin = (snapshotUrl: string) => {
     window.location.href = textUtil.sanitizeUrl(snapshotUrl);
-  }
-
-  render() {
-    const { isFullscreen, title, folderTitle } = this.props;
-    const onGoBack = isFullscreen ? this.onClose : undefined;
-
-    const titleHref = locationUtil.updateSearchParams(window.location.href, '?search=open');
-    const parentHref = locationUtil.updateSearchParams(window.location.href, '?search=open&folder=current');
+  };
 
-    return (
-      <PageToolbar
-        pageIcon={isFullscreen ? undefined : 'apps'}
-        title={title}
-        parent={folderTitle}
-        titleHref={titleHref}
-        parentHref={parentHref}
-        onGoBack={onGoBack}
-        leftItems={this.renderLeftActionsButton()}
-      >
-        {this.renderRightActionsButton()}
-      </PageToolbar>
-    );
-  }
-}
+  const { isFullscreen, title, folderTitle } = props;
+  // this ensures the component rerenders when the location changes
+  useLocation();
+  const titleHref = locationUtil.updateSearchParams(window.location.href, '?search=open');
+  const parentHref = locationUtil.updateSearchParams(window.location.href, '?search=open&folder=current');
```

Comment:
This looks strange / bad, updating url in render path? The function name for this util is really confusing :)  And it does something that is also pretty bad. 

This should be more like
```ts
const titleRef = locationSrv.getUrlForPartial({search: 'option'});
```

Reply:
@torkelo yeah i agree the naming of this function is bad since it doesn't update the actual url.
what it actually does is: given a url string and some search params you want to update, return a new url string.

as we're exposing this from `@grafana/data`, we're a little hamstrung by semver... could copy the method and rename it to `getUrlForPartial` and deprecate the old method until grafana 9? wdyt? 🤔

Reply:
yea, we could add a method, getUrlForPartial that takes in the location returned by useLocation? and does something similar to: https://github.com/grafana/grafana/blob/main/packages/grafana-runtime/src/services/LocationService.ts#L59  except just returns the new url